### PR TITLE
feat(_find_ml_task): Refine ML task detection logic

### LINKED ISSUE #1080: feat(train_test_split): Refine detection of ML task
### Is your feature request related to a problem? Please describe.

@koaning got the following error when running `skore.train_test_split`:
> /workspaces/joblib-modal/.venv/lib/python3.12/site-packages/skore/sklearn/train_test_split/train_test_split.py:165: HighClassImbalanceTooFewExamplesWarning: It seems that you have a classification problem with at least one class with fewer than 100 examples in the test set. In this case, using train_test_split may not be a good idea because of high variability in the scores obtained on the test set. We suggest three options to tackle this challenge: you can increase test_size, collect more data, or use skore's cross_validate function.

and wonders why skore is assuming that he is doing classification, while he is actually doing a regression.

@augustebaum points out that if the target vector contains a lot of integers, scikit-learn still interprets it as a classification problem.

### Describe the solution you'd like

If there are many integers in the target of a regression task, it should not be detected as a classification task.

@koaning suggests the following ideas:
- what about checking if the ints are sequential?
- And if it also includes a zero class?

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by augustebaum**
Closes #1080

Adds more logic in the case where `y` is detected as multiclass classification: we now additionally check whether `y` is a vector or sequential values that contains 0 (if so, we output "multiclass-classification", otherwise "regression"). This heuristic is suggested by @koaning.

### CODE REVIEW

File: skore/tests/unit/sklearn/test_utils.py [hunk 1]
```diff
@@ -53,6 +54,9 @@ def test_find_ml_task_with_estimator(X, y, estimator, expected_task, should_fit)
         (make_regression(n_samples=100, random_state=42)[1], "regression"),
         (None, "clustering"),
         (make_multilabel_classification(random_state=42)[1], "unsupported"),
+        (numpy.array([1, 5, 9]), "regression"),
+        (numpy.array([0, 1, 2]), "multiclass-classification"),
+        (numpy.array([1, 2, 3]), "regression"),
     ],
```
Comment (thomass-dev) [inline skore/tests/unit/sklearn/test_utils.py:60]:
```suggestion
        (numpy.array([0, 1, 5, 9]), "regression"),
    ],
```

File: skore/src/skore/sklearn/find_ml_task.py [hunk 2]
```diff
@@ -22,6 +32,22 @@ def _find_ml_task(y, estimator=None) -> MLTask:
...
         The guess of the kind of ML task being performed.
+
+    Examples
+    --------
+    >>> import numpy
+
+    # Discrete values, not sequential
+    >>> _find_ml_task(numpy.array([1, 5, 9]))
+    'regression'
+
```
Comment (thomass-dev) [inline skore/src/skore/sklearn/find_ml_task.py:43]: # Discrete values, not sequential, containing 0
    >>> _find_ml_task(numpy.array([0, 1, 5, 9]))
    'regression'

Comment (thomass-dev) [inline skore/src/skore/sklearn/find_ml_task.py:43]:
```suggestion
    # Discrete values, not sequential, containing 0
    >>> _find_ml_task(numpy.array([0, 1, 5, 9]))
    'regression'
    
```

**Review by thomass-dev — DISMISSED** [review]
